### ISSUE TITLE
feature request: omit 'not referred to' or inferable linear type binders

### ISSUE THREAD

**Issue by MarcelineVQ** [idris-lang/Idris2#175]
When I write something like a linear compose:
```idris
lindot : (1 f : (1 z : b) -> c) -> (1 g : (1 y : a) -> b) -> (1 x : a) -> c
lindot m n o = m (n o)
```
The type binders `z,y,x`, aren't things I should need to mention to write `lindot`. They clutter the type for the reader because now they need to keep in mind that there might be a `z`, `y`, or `x` later on in the definition, especially for some more complicated definition.

These binders currently need to be named because, in order to annotate the usage, we need a binder to assign that usage to. The syntax itself for that is fine but I should also be allowed to be explicit about what binder names I'm *not using directly in my definition*. In particular my implementation of lindot makes no mention of `z`,`y`,`x`, so neither should my type have to.

The feature request here is for the ability to write:
```
lindot : (1 f : (1 _ : b) -> c) -> (1 g : (1 _ : a) -> b) -> (1 _ : a) -> c
```
as my type. Mimicking the use of _ in patterns for "we don't care to name this".
This is not less verbose but it makes it explicit that there is no more specific use of _ binders than to assign them a usage.

A current solution for avoiding naming binders might look like
```idris
infixr 6 ~>
(~>) : Type -> Type -> Type
(~>) a b = (1 x : a) -> b

lindot : (1 f : b ~> c) -> (1 g : a ~> b) -> a ~> c
lindot m n o = m (n o)
```
This isn't ideal however because now we need an arrow for 0 and 1, bringing us to 3 function arrow types to cover our bases, which users then also need to know or look up. And one type signature might have all 3 arrows in it.

A person could even write (and the reason I used m n o above)
```idris 
lindot : (1 _ : (1 _ : b) -> c) -> (1 _ : (1 _ : a) -> b) -> (1 _ : a) -> c`
lindot m n o = m (n o)
```
with this proposal so good taste will still have to be used to determine what is readable to a user.  
This is still more readable to me than the version with all the binders named but regardless of whether that's the case for you I hope the usefulness of _ has been conveyed when it comes to having another tool to express our intent in our type signatures.

I hope this proposal/idea is clear but if not please do ask about it.

**Comment by chrrasmussen** [comment]
It seems like Edwin has solved this in 3120fcb84afec3c5e6d1e060d93c44cfe0f7b128. I did try to run your examples on the latest commit, and they all worked :)

**Comment by MarcelineVQ** [comment]
Shucks buster! Good job everyone :>
